pmd-apex/src/main/resources/category/apex/performance.xml

```diff
@@ -9,6 +9,39 @@
 Rules that flag suboptimal code.
     </description>
 
+    <rule name="AvoidDebugStatements"
+          language="apex"
+          since="6.36.0"
+          message="Avoid debug statements since they impact on performance"
+          class="net.sourceforge.pmd.lang.apex.rule.ApexXPathRule"
+          externalInfoUrl="${pmd.website.baseurl}/pmd_rules_apex_performance.html#avoiddebugstatements">
+        <description>
+Debug statements contibute to longer transactions and consume
+Apex CPU time even when there&apos;s no active debug trace.
```

Comment:
Is there something that could be recommended to do instead? E.g. long time before parameterized log messages where introduced in the java logging ecosystem, you usually used a guarding if clause around the logging (see [GuardLogStatement](https://pmd.github.io/latest/pmd_rules_java_bestpractices.html#guardlogstatement)). If I understand the rule correctly, this is a similar case here (e.g. creating a big string for the log message which is not written to the log because there's no active debug trace....).

By default, this rule says: never use `System.debug`. What should the user do instead?
We probably should at least mention that for valid cases, this rule needs to be suppressed (via annotation or NOPMD comment). Maybe even as a example?

I find your description in the issue #3307 very good and most information should be put here (because rule users won't read the issue, but read the rule documentation).

Reply:
I agree that the description needed quite more clarification, how about now?

Reply:
:+1: